apps/base-docs/docs/pages/identity/smart-wallet/guides/paymasters.mdx

```diff
@@ -1,52 +1,75 @@
+As a prerequisite, you'll need to obtain a paymaster service URL from a paymaster service provider.
+
+We recommend the [Coinbase Developer Platform](https://www.coinbase.com/developer-platform) paymaster,
+currently offering up to $15k in gas credits as part of the [Base Gasless Campaign](/identity/smart-wallet/introduction/base-gasless-campaign).
 
-We recommend the [Coinbase Developer Platform](https://www.coinbase.com/developer-platform) paymaster, currently offering up to $15k in gas credits as part of the Base Gasless Campaign(/identity/smart-wallet/introduction/base-gasless-campaign).
-You can find a full list of ERC-7677-compliant paymaster services [here](https://www.erc7677.xyz/ecosystem/paymasters).
+Once you have signed up for Coinbase Developer Platform, you get your Paymaster service URL by navigating to Onchain Tools > Paymaster as shown below:
+
+<div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center' }}>
+  <img
+    alt="Paymaster CDP"
+    src="/images/smart-wallet/PaymasterCDP.png"
+    width="1000"
+    loading="lazy"
+  />
+  <em>How to get your Paymaster service URL</em>
+</div>
 
 Once you choose a paymaster service provider and obtain a paymaster service URL, you can proceed to integration.
 
-### (Recommended) Setup your paymaster proxy
+<Callout type="warning">
+**ERC-7677-Compliant Paymaster Providers**
 
-Creating an API to proxy calls to your paymaster service is important for two reasons.
+To be compatible with Smart Wallet, the paymaster provider you choose must be [ERC-7677-compliant](https://www.erc7677.xyz/ecosystem/paymasters).
 
-1. Allows you to protect any API secret.
-2. Allows you to add extra validation on what requests you want to sponsor.
+</Callout>
 
-#### Validate UserOperation
 
-:::info
-`willSponsor` can be simplified or removed entirely if your paymaster service supports allowlisting which
-contracts and function calls should be sponsored. [Coinbase Developer Platform](https://www.coinbase.com/developer-platform) supports this.
-:::
-Before we write our proxy, let's write a `willSponsor` function to add some extra validation.
+### Validate UserOperation
+
 The policies on many paymaster services are quite simple and limited. As your API will be exposed on the web,
 you want to make sure in cannot abused: called to sponsor transaction you do not want to fund. The checks below
 are a bit tedious, but highly recommended to be safe. See "Trust and Validation" [here](https://hackmd.io/@AhweV9sISeevhvrtVPCGDw/BynRsX7Ca#Trust-and-Validation)
 for more on this.
 
+The goal of this section is to write a `willSponsor` function to add some extra validation if needed.
+
+<Callout type="success">
+**About `willSponsor`**
+
+`willSponsor` can be simplified or removed entirely if your paymaster service supports allowlisting which
+contracts and function calls should be sponsored. [Coinbase Developer Platform](https://www.coinbase.com/developer-platform) supports this.
```

Comment:
here we should simplify and add opinion by saying something like "willSponsor is only necessary for advanced use cases that require logic beyond CDP custom limits <link to resource or further explanation of configurable limits>"